src/main/java/com/example/petbuddybackend/service/user/ClientService.java

```diff
@@ -47,6 +55,34 @@ public void createClientIfNotExist(JwtAuthenticationToken token) {
         }
     }
 
+    public ClientDTO getClient(String clientEmail) {
+        Client client = getClientByEmail(clientEmail);
+        userService.renewProfilePicture(client.getAccountData());
+        return clientMapper.mapToClientDTO(client);
+    }
+
+    public ClientComplexInfoDTO addFollowingCaretakers(String clientEmail, Set<String> caretakerEmails) {
+
+        Client client = getClientByEmail(clientEmail);
+        assertClientNotFollowingItself(client, caretakerEmails);
+        assertNewFollowingCaretakersAreNotAlreadyFollowed(client, caretakerEmails);
+
+        Set<Caretaker> caretakersToFollow = getCaretakersByEmails(caretakerEmails);
+        client.getFollowingCaretakers().addAll(caretakersToFollow);
+        return clientMapper.mapToClientComplexInfoDTO(clientRepository.save(client));
+    }
+
+    public ClientComplexInfoDTO removeFollowingCaretakers(String clientEmail, Set<String> caretakerEmails) {
+
+        Client client = getClientByEmail(clientEmail);
+        assertClientNotFollowingItself(client, caretakerEmails);
+        assertCaretakersToRemoveAreFollowed(client, caretakerEmails);
```

Comment:
Tu podobnie. Jak chcemy usunąć kogoś kogo nie ma to wydaje mi się, że API może zwrócić OK

Reply:
Tutaj tez bym zwrocil info jesli unfollowujemy kogos kogo nie followujemy, chociaz i tak front nie dopusci pewnie do tego